components/playlist/resources/reducers/playlist_reducer.ts

```diff
@@ -12,14 +12,27 @@ import * as PlaylistMojo from 'gen/brave/components/playlist/mojom/playlist.mojo
 const playlistReducer: Reducer<Playlist.State|undefined> =
     (state: Playlist.State|undefined, action) => {
       if (state === undefined) {
-        state = { lists: [] }
+        state = { lists: [], currentList: undefined }
       }
 
       // TODO(sko) Handle each action properly.
       switch (action.type) {
         case types.PLAYLIST_LOADED:
-          state = { ...state, lists: [...action.payload as PlaylistMojo.Playlist[]] }
+          const playlists = action.payload as PlaylistMojo.Playlist[]
+          let currentList
+          if (state.currentList) { currentList = playlists.find(list => list.id === state!.currentList!.id) }
+          if (!currentList) { currentList = playlists[0] }
+
+          if (playlists.length && !currentList) {
+            console.error('there\'s no selected playlist eventhough we have playlists')
+          }
+
+          state = { ...state, currentList, lists: [...playlists] }
           break
+
+        case types.PLAYLIST_SELECTED:
+          const playlist = action.payload as PlaylistMojo.Playlist
```

Comment:
I'm curious - doesn't Typescript infer the type of `action.payload` here? If so we can get rid of the cast.

Reply:
This too. I was too familiar to typed language and kinda confused by payload type during dev haha 😹 So I left `as ...` for me. Removed!

Reply:
Yeah, it's definitely an interesting switch to Typescript 😹It does a bunch of really cool inference, which can definitely seem like magic sometimes 😆